lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -834,20 +834,19 @@ _maxScriptExecutionCost
     :: ProtocolParameters
     -> [Redeemer]
     -> Coin
-_maxScriptExecutionCost pp (length -> numberOfScripts)  =
-    case view #executionUnitPrices pp of
-        Just prices -> totalCost $ executionCost prices maxExecutionUnits
+_maxScriptExecutionCost pp redeemers
+    | not (null redeemers) = case view #executionUnitPrices pp of
+        Just prices -> executionCost prices maxExecutionUnits
         Nothing     -> Coin 0
+    | otherwise            = Coin 0
```

Comment:
Perhaps don't align this here? I don't think this really helps with readability, as the thing it's aligning with is actually separated over several lines (by a completely separate pattern match).

```suggestion
    | otherwise = Coin 0
```